src/Tools/IiifUrlValidator.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: dpino
+ * Date: 11/27/18
+ * Time: 1:06 PM
+ */
+
+namespace Drupal\format_strawberryfield\Tools;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\ServerException;
+use GuzzleHttp\Exception\ConnectException;
+
+class IiifUrlValidator {
+
+  use StringTranslationTrait;
+
+  /**
+   * @var \GuzzleHttp\Client;
+   */
+  protected $httpClient;
+
+  public function __construct () {
+    $this->httpClient = $this->httpClient = \Drupal::httpClient();
+  }
+
+  public function checkPublicUrl($publicUrl) {
+    // check if there's a value entered
+    if (trim($publicUrl)) {
+      $isLocalhost = strpos($publicUrl, 'localhost');
+      if ($isLocalhost !== false) {
```

Comment:
This is hard to read. Try to use the Elvis Operator more and less nested if and else, like prepare your values before doing the ifs and else. Sometimes it is not possible of course, but here you can.
https://stackoverflow.com/questions/1993409/operator-the-elvis-operator-in-php